### ISSUE TITLE
Whats is the proper way to set a relation column to None ? 

### ISSUE THREAD

**Issue by hakib** [josiahcarlson/rom#56]
```python
import rom

class User(rom.Model):
    name = rom.String()
    group = rom.ManyToOne('Group')

class Group(rom.Model):
    name = rom.String()
    users = rom.OneToMany('User', on_delete='cascade')

g = Group(name='dev')
g.save()
u = User(name='haki', group=g)
u.save()

u.group
 <__main__.Group at 0x438f358>

u.group = None
\venv\lib\site-packages\rom\columns.pyc in __set__(self, obj, value)
    219                 value = self._from_redis(value)
    220         except (ValueError, TypeError):
--> 221             raise InvalidColumnValue("Cannot convert %r into type %s"%(value, self._allowed))
    222         self._validate(value)
    223         obj._data[self._attr] = value

InvalidColumnValue: Cannot convert None into type ()
```
I've tried setting it to an empty tuple \ array - same result.
also noticed that the `_alowed` array is empty so i'm not sure what i can use to represent an empty relation. 

New objects are created with an empty relation 
```python
u2 = User(name='No group')

u2.save()
3

u2.group
<<nothing>> 
u2
<__main__.User at 0x4379358>
```

Thanks.

**Comment by josiahcarlson** [comment]
Right now you should be able to: `del u.group`, but you should definitely also be able to set an attribute to None as an alternative to deleting the attribute directly. I'll add that functionality some time in the coming days and put out a new release.

Thank you for the report!

**Comment by hakib** [comment]
Thanks @josiahcarlson, I've tried it and it seems that deleting the attribute prevents me from setting it later on in the session. I'm currently setting it to a non existing key, e.g `u.group = -1` and it evaluates to none without removing the attr. I haven't dug deep into it but my guess is that a relation cannot be found, the __get__ failes gracefully and returns None.

BTW,  I'm not sure it deserves a separate thread so i'll just ask here - Is it possible to provide my own value to a pk field ?

**Comment by josiahcarlson** [comment]
Finally getting around to fixing this.

Yes, the `.get()` on a missing entity (your assigning the value to -1) does result in graceful failover to returning None.

Can you provide your own value to a primary key column? Yes, but not without some work (one extra line). Looking at your other bug, you ended up using a secondary string column instead of an integer - that may have strange behavior when it comes to some search features, which I'll try to explain in the other bug.

**Comment by josiahcarlson** [comment]
Fixed in release 0.29.4, out now.

**Comment by hakib** [comment]
Hey @josiahcarlson, 

When setting an already None column to None it throws a value error.

``` python
def test_rom_set_column_none(self):
    
    class SimpleModel(rom.Model):
      s = rom.String()

    x = SimpleModel()
    self.assertEqual(x.s, None)
    x.s = "string"
    self.assertEqual(x.s, "string")
    x.s = None
    self.assertIsNone(x.s)
    x.s = None # <-- AttributeError
    self.assertIsNone(x.s)
```

Relevant code is [here](https://github.com/josiahcarlson/rom/blob/master/rom/columns.py#L241). The first statment sets `"string"` in `SimpleModel._data['s']`, The second assignment sets `s` to `None` which calls delete and pops `s` from `_data`. The second none assignmnet encounters a `KeyError` and treats `s` as a non existing column.

0.29.5

**Comment by josiahcarlson** [comment]
I have reproduced the issue and have a fix for it.

Can you confirm and/or comment on #58 ?

**Comment by hakib** [comment]
I tried too but all my tests failed on this issue. I will try and work
around it.
בתאריך 30 במרץ 2015 19:00, "Josiah Carlson" <notifications@github.com> כתב:

**Comment by josiahcarlson** [comment]
> I tried too but all my tests failed on this issue. I will try and work
around it.

I am trying to understand what you are saying, but I don't have enough context to understand.

Have you read my most recent comment on #58 ? What have you tried? What doesn't work? What are you trying to do?

**Comment by hakib** [comment]
Sorry, Ill try to be more clear. 

I started going over your comments for #58 after upgrading to rom 0.29.5. I was unable to run my tests because they failed on this issue (setting none to a none field). I will try to provide more context on the other issue as well.

**Comment by josiahcarlson** [comment]
Okay. Let me be clearer then.

There does exist an issue in rom 0.29.5 for setting an already empty/`None` column to `None`. I have reproduced it. I have a fix for it, but that fix requires another release, unless you want to catch the `AttributeError` yourself.

**Comment by josiahcarlson** [comment]
The AttributeError should be fixed in 0.30.0 .
